plugins/woocommerce/src/Internal/Admin/RemoteFreeExtensions/GetRuleProcessor.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Automattic\WooCommerce\Internal\Admin\RemoteFreeExtensions;
+
+use Automattic\WooCommerce\Admin\RemoteSpecs\RuleProcessors\GetRuleProcessor as OriginalGetRuleProcessor;
+use Automattic\WooCommerce\Admin\RemoteSpecs\RuleProcessors\RuleProcessorInterface;
+use Automattic\WooCommerce\Internal\Admin\RemoteFreeExtensions\Rules\ContextPluginsRuleProcessor;
+use Automattic\WooCommerce\Internal\Admin\RemoteFreeExtensions\Rules\ContextVarsRuleProcessor;
+
+/**
+ * A custom GetRuleProcessor class to support context_vars and context_plugins rule types.
+ *
+ * GetRuleProcessor class.
+ */
+class GetRuleProcessor {
```

Comment:
I feel the name of this could be more descriptive. `Perhaps GetRuleProcessorForContext` 🤔

Reply:
@adrianduffell Good suggestion! 👍

Reply:
@adrianduffell Updated it in [Rename rule processor](https://github.com/woocommerce/woocommerce/pull/56233/commits/76fe21ca5ad6b00c9d26ee4a26dd51293d8e8b31)